src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/DefaultSemanticTokensColorsProvider.java

```diff
@@ -29,6 +29,12 @@ public class DefaultSemanticTokensColorsProvider implements SemanticTokensColors
     public @Nullable TextAttributesKey getTextAttributesKey(@NotNull String tokenType,
                                                             @NotNull List<String> tokenModifiers,
                                                             @NotNull PsiFile file) {
+        // If this is for a custom semantic token type that expresses an inheritance relationship, swap it out now

```

Comment:
This type of logic will exist anywhere we're implementing behavior based on a known semantic token type. It allows a custom semantic token type to base its behavior on a standard semantic token type.